lib/features/email/presentation/controller/single_email_controller.dart

```diff
@@ -753,6 +785,24 @@ class SingleEmailController extends BaseController with AppLoaderMixin {
         success.attachment.generateFileName());
   }
 
+  void _viewAttachmentForWebSuccessAction(
+    ViewAttachmentForWebSuccess success,
+  ) {
+    log('SingleEmailController::_viewAttachmentForWebSuccessAction():');
+    final mimeType = success.attachment.type?.mimeType ??
+        lookupMimeType('', headerBytes: success.bytes);
+    if (mimeType != Constant.pdfMimeType) {
+      _downloadAttachmentForWebSuccessAction(success);
+      return;
+    }
+
+    mailboxDashBoardController.deleteDownloadTask(success.taskId);
+
+    _downloadManager.openDownloadedFileWeb(
+        success.bytes, 
+        success.attachment.type?.mimeType);
```

Comment:
Why not separate actions for `view` and `download` right from clicking on the attachment item in the email view screen. As far as I remember, every attachment that appears in the email view already has a `mimeType`.